### PR TITLE
book listings DAO finished

### PR DESCRIPTION
**Pull request by Aidan4478D**
create a listing by sending a POST request to http://localhost:8080/listings/create where the body is of the form

{
    "userId": 1,
    "bookId": 1,
    "listingStatus": "new"
}

**make sure that the userId and bookId exist** in the users and book_listings table though or else you'll get an error

update a listing by sending a POST request to http://localhost:8080/listings/update/{id} where the body is of the same form as creating a listing

delete a listing by sending a GET request to http://localhost:8080/listings/delete/{id} where id is the listing_id you want to delete

can check for book count by sending a GET request to http://localhost:8080/listings/count 
and view an individual listing by sending a GET request to http://localhost:8080/listings/{id} where again id is the listing_id you want to delete

### CODE REVIEW

**Comment by Aidan4478D** [comment]
should hopefully not run into any merge conflicts as this listingsDAO branch is already merged with the updatecurrentDAO branch and with main

File: README.md [hunk 1]
```diff
@@ -69,9 +96,15 @@ The project goal is to create a simple marketplace for book lovers, while not ha
...
+        - URL: http://localhost:8080/users/delete/{id}
+
     - Count books:
         - URL: http://localhost:8080/books/count
 
     - Get book by ID:
         - URL: http://localhost:8080/books/{id}
 
+    - Delete book by ID:
+        - URL: http://localhost:8080/books/delete/{id}
```
Comment (Yichi-liu) [inline README.md:109]: maybe add a README so its easier to test it later

File: init-scripts/init_db.sql [hunk 2]
```diff
@@ -12,7 +12,8 @@ CREATE TABLE IF NOT EXISTS users (
 CREATE TABLE IF NOT EXISTS book_info (
   book_id SERIAL PRIMARY KEY,
   title VARCHAR(255) NOT NULL,
-  isbn INT NOT NULL UNIQUE,
+  --isbn BIGINT NOT NULL UNIQUE,
+  isbn NUMERIC(13, 0) NOT NULL UNIQUE,
```
Comment (Yichi-liu) [inline init-scripts/init_db.sql:16]: nice

File: src/main/java/com/thebookcooper/controller/BookInfoController.java [hunk 3]
```diff
@@ -82,4 +82,44 @@ public Book getBookById(@PathVariable("id") long id) {
         }
         return null; // Or return an appropriate response/entity indicating not found or error
     }
+
+    @PostMapping("/books/update/{id}")
```
Comment (Yichi-liu) [inline src/main/java/com/thebookcooper/controller/BookInfoController.java:86]: putmapping

File: src/main/java/com/thebookcooper/controller/BookInfoController.java [hunk 4]
```diff
@@ -82,4 +82,44 @@ public Book getBookById(@PathVariable("id") long id) {
...
+
+            //calls update function from dao/BookInfoDAO to update listing
+            return infoDAO.update(updatedBook);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException("Failed to update the book", e);
+        }
+    }
+
+    @GetMapping("/books/delete/{id}")
```
Comment (Yichi-liu) [inline src/main/java/com/thebookcooper/controller/BookInfoController.java:114]: deletemapping

File: src/main/java/com/thebookcooper/controller/ListingsController.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+            return listDAO.update(updatedListing);
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException("Failed to update the listing", e);
+        }
+    }
+
+    @GetMapping("/listings/delete/{id}")
```
Comment (Yichi-liu) [inline src/main/java/com/thebookcooper/controller/ListingsController.java:95]: deletemapping

File: src/main/java/com/thebookcooper/controller/ListingsController.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                return "Number of listings: " + resultSet.getInt(1);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return "Error retrieving book count";
+        }
+        return "No listings found";
+    }
+
+    @PostMapping("/listings/update/{id}")
```
Comment (Yichi-liu) [inline src/main/java/com/thebookcooper/controller/ListingsController.java:70]: putmapping

**Review by Yichi-liu — APPROVED** [review]

**Pull request merged**